### PR TITLE
Use context.WithoutCancel in azbus

### PR DESCRIPTION
**Pull request by eccles**
The handcrafted contextWithoutCancel is replaced with correct standard context.WithoutCancel from Go 1.21.

If processing a message takes longer than the message deadline a WARNING is emitted with suggested remedial action.

Type aliases in aliases.go have been moved closer to their point of usage and aliases.go deleted.

Internal logger is indexed with the name of the sender/receiver.

Mock generation Taskfile is added.

AB#8276

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ replace github.com/rkvst/go-rkvstcommon => ./
 
 require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.1
-	github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus v1.2.1
+	github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus v1.4.0
```
Comment (robinbryce) [inline go.mod:9]: is this version bump intentional ?

File: azbus/receiver.go [hunk 2]
```diff
@@ -128,12 +136,25 @@ func (r *Receiver) String() string {
...
+	log := r.log.FromContext(ctx)
+	defer log.Close()
+
+	log.Debugf("Processing message %d of %d", count, total)
 	err := handler.Handle(ctx, msg)
-	r.log.Debugf("%s: Processing message took %s", r, time.Since(now))
+	duration := time.Since(now)
+	log.Debugf("Processing message %d took %s", count, duration)
+	// This is safe because maxDuration is only defined if RenewMessageLock is false.
+	if !r.Cfg.RenewMessageLock && duration >= maxDuration {
```
Comment (robinbryce) [inline azbus/receiver.go:150]: Great helpful telemetry, thanks

**Review by robinbryce — CHANGES_REQUESTED** [review]
If you can confirm the version bump is intentional I will tick, thanks

File: go.mod (hunk 1 above)
Comment (eccles) [reply go.mod:9]: yes - we dont get the CodeUnauthorizedAccees in errors.go otherwise

**Review by robinbryce — APPROVED** [review]

**Pull request merged**